prereise/gather/griddata/transmission/geometry.py

```diff
@@ -0,0 +1,273 @@
+from dataclasses import dataclass
+from math import exp, log, pi, sqrt
+from statistics import geometric_mean
+
+from prereise.gather.griddata.transmission.const import (
+    epsilon_0,
+    mu_0,
+    relative_permeability,
+    resistivity,
+)
+
+
+@dataclass
+class Conductor:
+    """Represent a single conductor (which may be a stranded compsite).
+
+    :param float radius: outer radius of conductor.
+    :param str material: material of conductor. Used to calculate ``resistance_per_km``
+        and ``gmr`` if these aren't passed to the constructor, unnecessary otherwise.
+    :param float resistance_per_km: outer radius of conductor. Will be estimated from
+        other parameters if it isn't passed.
+    :param float gmr: geometric mean radius of conductor. Will be estimated from
+        other parameters if it isn't passed.
+    :param float area: cross-sectional area of conductor. Will be estimated from
+        other parameters if it isn't passed.
+    """
+
+    radius: float
+    material: str = None
+    resistance_per_km: float = None
+    gmr: float = None
+    area: float = None
+
+    def __post_init__(self):
+        # Validate inputs
+        if self.gmr is None and (self.material is None or self.radius is None):
+            raise ValueError(
+                "If gmr is not provided, material and radius are needed to estimate"
+            )
+        if self.resistance_per_km is None and (
+            self.material is None or self.radius is None
+        ):
+            raise ValueError(
+                "If resistance_per_km is not provided, "
+                "material and radius are needed to estimate"
+            )
+
+        if self.gmr is None:
+            try:
+                self.permeability = relative_permeability[self.material]
+            except KeyError:
+                raise ValueError(
+                    f"Unknown permeability for {self.material}, can't calculate gmr"
+                )
+            self.gmr = self.radius * exp(self.permeability / 4)
+
+        if self.resistance_per_km is None:
+            try:
+                self.resistivity = resistivity[self.material]
+            except KeyError:
+                raise ValueError(
+                    f"Unknown resistivity for {self.material}, "
+                    "can't calculate resistance"
+                )
+            if self.area is None:
+                self.area = pi * self.radius**2
```

Comment:
`self.area` is useful only if `self.resistance_per_km` is not provided and needs to be calculated? Not sure whether we should put this up to the same level of other attributes check.

Reply:
We always have `radius`, so we can always estimate the cross-sectional area if necessary (as long as we also have the `material`, which we already check for).